metricbeat/helper/prometheus/prometheus.go

```diff
@@ -99,14 +116,23 @@ func (p *prometheus) GetFamilies() ([]*dto.MetricFamily, error) {
 		return nil, fmt.Errorf("unexpected status code %d from server", resp.StatusCode)
 	}
 
+	buf := byteBuffer.Get().(*bytes.Buffer)
+	buf.Reset()
+	defer byteBuffer.Put(buf)
+
+	written, err := io.Copy(buf, io.LimitReader(reader, int64(p.config.MaxContentSize)))
```

Comment:
WDYT about using an `http.MaxBytesReader`? I think we may not need the buffer if we use that one.